src/app/clusters/actions-server/actions-server.h

```diff
@@ -0,0 +1,370 @@
+/*
+ *
+ *    Copyright (c) 2024 Project CHIP Authors
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+#include <app-common/zap-generated/cluster-objects.h>
+#include <app-common/zap-generated/ids/Clusters.h>
+#include <app/AttributeAccessInterface.h>
+#include <app/CommandHandlerInterface.h>
+#include <app/reporting/reporting.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+namespace Clusters {
+namespace Actions {
+
+static constexpr size_t kActionNameMaxSize       = 128u;
+static constexpr size_t kEndpointListNameMaxSize = 128u;
+
+static constexpr size_t kEndpointListMaxSize = 256u;
+
+class Delegate;
+
+struct ActionStructStorage : public Structs::ActionStruct::Type
+{
+    ActionStructStorage(){};
+
+    ActionStructStorage(uint16_t aAction, const CharSpan & aActionName, ActionTypeEnum aActionType, uint16_t aEpListID,
+                        BitMask<CommandBits> aCommands, ActionStateEnum aActionState)
+    {
+        Set(aAction, aActionName, aActionType, aEpListID, aCommands, aActionState);
+    }
+
+    ActionStructStorage(const ActionStructStorage & aAction) { *this = aAction; }
+
+    ActionStructStorage & operator=(const ActionStructStorage & aAction)
+    {
+        Set(aAction.actionID, aAction.name, aAction.type, aAction.endpointListID, aAction.supportedCommands, aAction.state);
+        return *this;
+    }
+
+    void Set(uint16_t aAction, const CharSpan & aActionName, ActionTypeEnum aActionType, uint16_t aEpListID,
+             BitMask<CommandBits> aCommands, ActionStateEnum aActionState)
+    {
+        actionID          = aAction;
+        type              = aActionType;
+        endpointListID    = aEpListID;
+        supportedCommands = aCommands;
+        state             = aActionState;
+        MutableCharSpan mActionName(mBuffer);
+        CopyCharSpanToMutableCharSpanWithTruncation(aActionName, mActionName);
+        name = mActionName;
```

Comment:
```suggestion
        MutableCharSpan actionName(mBuffer);
        CopyCharSpanToMutableCharSpanWithTruncation(aActionName, actionName);
        name = actionName;
```
It's not a member, so no "m" prefix...